### ISSUE TITLE
npm install fails when using Nexus proxy

### ISSUE THREAD

**Issue by vadimanisimov** [npm/npm#7896]
Hi,

I'm trying to use npm behind proxy. To reduce duplicate downloads I've decided to setup Nexus proxy for npm: https://books.sonatype.com/nexus-book/reference/npm-proxying-registries.html#fig-npm-proxy
Unfortunately, 'npm install bower' fails with error:

```
2191 http fetch GET http://repo/content/repositories/npmjs/semver/-/semver-2.3.2.tgz
2192 http fetch 500 http://repo/content/repositories/npmjs/archy/-/archy-1.0.0.tgz
2193 error fetch failed http://repo/content/repositories/npmjs/archy/-/archy-1.0.0.tgz
2194 http fetch 500 http://repo/content/repositories/npmjs/semver/-/semver-2.3.2.tgz
2195 error fetch failed http://repo/content/repositories/npmjs/semver/-/semver-2.3.2.tgz
2196 verbose about to build C:\Users\username\node_modules\bower
2197 verbose unlock done using C:\Users\username\AppData\Roaming\npm-cache\_locks\bower-0fc0e1328a822957.lock for C:\Users\username\node_modules\bower
2198 verbose stack Error: fetch failed with status code 500
2198 verbose stack     at Request.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\npm-registry-client\lib\fetch.js:51:14)
2198 verbose stack     at Request.emit (events.js:107:17)
2198 verbose stack     at Request.onRequestResponse (C:\Program Files\nodejs\node_modules\npm\node_modules\request\request.js:1176:10)
2198 verbose stack     at ClientRequest.emit (events.js:107:17)
2198 verbose stack     at HTTPParser.parserOnIncomingClient (_http_client.js:426:21)
2198 verbose stack     at HTTPParser.parserOnHeadersComplete (_http_common.js:111:23)
2198 verbose stack     at Socket.socketOnData (_http_client.js:317:20)
2198 verbose stack     at Socket.emit (events.js:107:17)
2198 verbose stack     at readableAddChunk (_stream_readable.js:163:16)
2198 verbose stack     at Socket.Readable.push (_stream_readable.js:126:10)
```

I haven't paste here whole npm_debug.log, because it has more that 3000 rows

If I set registry to https://registry.npmjs.org and set local http_proxy, npm install works fine.

Does anyone have any idea about this issue?

**Comment by smikes** [comment]
No, I'm sorry, I'm not familiar with the Nexus programs.  Have you checked Sonatype's documentation / knowledge base?

**Comment by vadimanisimov** [comment]
Hi,

Yes, I've configured Nexus repo accordingly Sonatype's documentation. They provide 2 types of npm repos (hosted and proxy). I've used proxy repo and set npm registry value to this local repo URL.

It's interesting, that many packages were installed:
1836 http fetch GET http://repo/content/repositories/npmjs/tar-fs/-/tar-fs-1.5.0.tgz
1837 http fetch 200 http://repo/content/repositories/npmjs/shell-quote/-/shell-quote-1.4.3.tgz
1838 http fetch 200 http://repo/content/repositories/npmjs/rimraf/-/rimraf-2.3.2.tgz
1839 http fetch 200 http://repo/content/repositories/npmjs/mkdirp/-/mkdirp-0.5.0.tgz

But some of them thrown error (I've pasted log in my first post)

**Comment by vadimanisimov** [comment]
Hi,

Issue was on Sonatype side.
Thank you for your time.

**Comment by mdurocher** [comment]
Hi,

I am facing the same problem, that is a random error 500 when retrieving npm stuff from Nexus. 
Do you know which version of Nexus corrects the bug ? 
I am a bit confused because I use Nexus 2.11.06 in which the problem is supposed to be fixed.

**Comment by vadimanisimov** [comment]
Hi,

This bug was in 2.10 and was fixed in 2.11.
But if some packages were created, while you were using 2.10, they were not fixed during update.
To fix this issue I had to manually remove packages, mentioned in 'error 500' messages.

I hope you have same root cause as I had and this will help you to fix the problem.

**Comment by mdurocher** [comment]
Thanks for the fast response !
Unfortunately, no packages where created while using previous version ...
Error 500 is random, not specific to a package, so the cause seems to be elsewhere.
